File: SW Expert Academy/D2/01288. 새로운 불면증 치료법/새로운 불면증 치료법.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
	int T; cin >> T;
	for (int tc = 0; tc < T; tc++) {
		printf("#%d ", tc + 1);

		int N, num; cin >> N; num = N;
		vector<bool> count(10, false);
		while (1) {
			int tmp = N;
			while (tmp) {
				count[tmp % 10] = true;
				tmp /= 10;
			}

			bool flag = true;
			for (int i = 0; i < 10; i++) {
				if (!count[i]) {
					flag = false;
					break;
				}
			}

			if (flag) {
				printf("%d\n", N);
				break;
			}
			else N += num;
		}
	}
}

File: SW Expert Academy/D2/01288. 새로운 불면증 치료법/새로운 불면증 치료법.py
for tc in range(int(input())):
    print("#%d"%(tc+1), end=' ')

    N = int(input())
    num = N
    count = [False for i in range(10)]
    while True:
        tmp = N
        while tmp > 0:
            count[tmp%10] = True
            tmp //= 10

        flag = True
        for i in range(10):
            if not count[i]:
                flag = False
                break

        if flag:
            print(N)
            break
        else: N += num
